### ISSUE TITLE
Workspace - Creation of Free WS in OD by new acc doesn't redirect to WS settings page

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#36270]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.39-0
**Reproducible in staging?:** y
**Reproducible in production?:** n
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** applause internal team
**Slack conversation:**

## Action Performed:
Precondition: Logout from NewDot.

1. Go to OldDot (stage.expensify.com) and sign-up for a NEW account (don't validate it yet)
2. Navigate to Settings > Workspaces > Group > New Workspace > Select Free
## Expected Result:
User is redirected to NewDot > Workspace settings page.

## Actual Result:
User is redirected to NewDot > WS chat.

## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/43996225/a230a72f-5390-4b9b-a2dc-9f3dbb0c3441

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by kavimuru** [comment]
We think that this might be related to #wave5-free-submitters

**Comment by jeremy-croff** [comment]
I'm unable to reproduce because signing up on OD takes me straight to ND. Even after forcing ND, it won't reproduce because the problem seems to be related with fetching the user info and then navigating.

**Comment by stitesExpensify** [comment]
@jeremy-croff you need to use a non-public domain to test this unfortunately

**Comment by jeremy-croff** [comment]
> @jeremy-croff you need to use a non-public domain to test this unfortunately

If there is any process to get access let me know. If not, should this be external? We have to be able to reproduce to be able to propose/fix.

**Comment by marcaaron** [comment]
The main lead I have for this one is that the user gets navigated to this route:

![2024-02-09_12-43-05](https://github.com/Expensify/App/assets/32969087/cc6d8514-ac3d-4dbb-90ee-48747379747a)

Which looks sus because it's a workspace route with a report route in it. I'm not sure if that's something we support or not.

**Comment by marcaaron** [comment]
Confirmed it does not happen in production. 

p.s. a workaround for the non-public domain issue is to add the `?exitTo=inbox` param when you sign up for an account on OldDot.

**Comment by marcaaron** [comment]
Here's what I am observing...

By the time we get [here](https://github.com/Expensify/App/blob/22d36638391540d6f84e1ebe419f7aa8206c6876/src/libs/actions/Welcome.ts#L134-L139) we have already rendered the `workspace/overview` page and it's the current route. There's no "transition" route at all.

I couldn't figure out much about the `/w/` route. But it seems related to [this code](https://github.com/Expensify/App/blob/22d36638391540d6f84e1ebe419f7aa8206c6876/src/libs/Navigation/linkTo.ts#L131-L147).

What's clear is that we should not be redirecting to the workspace chat [here](https://github.com/Expensify/App/blob/22d36638391540d6f84e1ebe419f7aa8206c6876/src/libs/actions/Welcome.ts#L159). So, we can fix it by looking to see if the active route is the `/workspace/:policyID/overview`. 

Which sounds fine for a temporary solution... curious to hear @adamgrzybowski's thoughts for a potentially longer term/better solution.

cc @mountiny @hayata-suenaga as well.

**Comment by Beamanator** [comment]
Marking NAB since this got to staging

**Comment by hayata-suenaga** [comment]
Thank you Marc for bringing this to our attention. As @adamgrzybowski worked on the navigation for the Ideal Nav project, I'll ask them for their advice and feedback.

**Comment by adamgrzybowski** [comment]
@marcaaron 
> Which looks sus because it's a workspace route with a report route in it. I'm not sure if that's something we support or not.

This is supported

About the problem. Do you have an idea how I can reproduce this issue so I can see what's going on inside?

Also cc: @WojtekBoman as he was working on the workspace switcher

**Comment by marcaaron** [comment]
Yes, you can revert the change here and perform the tests steps in https://github.com/Expensify/App/pull/36289

**Comment by adamgrzybowski** [comment]
> Yes, you can revert the change here and perform the tests steps in https://github.com/Expensify/App/pull/36289

The olddot redirects me to `staging.new.expensify.com` or `new.expensify.com`. Is there a way to change it to the `dev`?

**Comment by mountiny** [comment]
@adamgrzybowski you need to intercept the route and change the url to point to your local dev. So you need to be quick in copying the transition link.

@koko57 had been working on this before, maybe she has some tips

**Comment by WojtekBoman** [comment]
I've reproduced this issue and now I'm investigating it :)

**Comment by WojtekBoman** [comment]
Maybe we could add an additional check to this if statement, for example:

```
   useEffect(() => {
        const navigationState = navigation.getState();
        const routes = navigationState.routes;
        const currentRoute = routes[navigationState.index];
        const bottomTabRoute = getTopmostBottomTabRoute(navigationState);
        if (
            currentRoute &&
            currentRoute.name !== NAVIGATORS.BOTTOM_TAB_NAVIGATOR &&
            // additional check
            (currentRoute.name !== NAVIGATORS.CENTRAL_PANE_NAVIGATOR || bottomTabRoute?.name === SCREENS.WORKSPACE.INITIAL)
        ) {
            return;
        }

        Welcome.show(routes, () => Navigation.navigate(ROUTES.ONBOARD));
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [isLoadingApp]);
```
Thanks to it, we're able to detect if we are currently on the settings tab and prevent unnecessarily calling `Welcome.show()`

**Comment by WojtekBoman** [comment]
Additionally, I've verified that the current fix doesn't prevent redirection when a tab is changed:

https://github.com/Expensify/App/assets/47774969/ff83b19a-b525-4bb9-94c6-2ab0fca84003

**Comment by mountiny** [comment]
@WojtekBoman thanks! would you like to create a PR with testing steps?

**Comment by WojtekBoman** [comment]
> @WojtekBoman thanks! would you like to create a PR with testing steps?

Here's [the PR](https://github.com/Expensify/App/pull/36498)

**Comment by marcaaron** [comment]
@WojtekBoman Nice!

**Comment by miljakljajic** [comment]
Any action left for this one? I see the PR was merged successfully and that payment isn't required - if we can close, go ahead!
